style(replays): added media query to the replay index so the table looks more sensible on smaller screens and corrected dark theme background

Danondso opened the pull request:
# Description
Closes #33780
Closes #33836 (it's a one-liner style fix so I included it in here)
- Fixed an issue where the sort icon was wrapping for TimeStamp.

#Screenshots 
Wide Screen: 
<img width="1686" alt="image" src="https://user-images.githubusercontent.com/7014871/164559713-5c38f4db-13c6-432c-b602-093e7139d5e3.png">
Slightly Shrunken:
<img width="1499" alt="image" src="https://user-images.githubusercontent.com/7014871/164559745-cd7f3a80-9611-4744-ac21-440682824c94.png">
Around 1200: 
<img width="1269" alt="image" src="https://user-images.githubusercontent.com/7014871/164559794-8d22a7a7-3fc8-4a69-9b2b-6a4c8422e17c.png">
Small Screens (this shows the updated vertical alignment for timestamp and project)
<img width="683" alt="image" src="https://user-images.githubusercontent.com/7014871/164792057-04bfd0ba-9d3b-4f36-997f-e57260bd3637.png">

Danondso committed:
style: updated the media query for the table so widths make sense for smaller screens

Danondso committed:
fix: dark theme background was incorrect on the replays screen.

billyvg commented:
* Why is timestamp missing on the 1200px width one?
* Can we make sure the timestamp doesn't wrap? it's making the row taller and un-centered.

ryan953 commented:
+1 on the 1200px rendering. I think truncating the path is better than having side-scrolling. Paths can get really long, I think truncating is inevitable.

ryan953 commented on static/app/views/replays/replays.tsx:
```diff
@@ -215,7 +215,13 @@ const StyledPageHeader = styled(PageHeader)`
+
+const StyledPanelTable = styled(PanelTable)`
+  @media (max-width: ${p => p.theme.breakpoints[1]}) {
+    grid-template-columns: repeat(3, minmax(0, 1fr));
```
instead of `1fr` would this work better, at all breakpoints, with something like `grid-template-columns: auto max-content max-content;`?

Jesse-Box replied:
@Danondso @ryan953 

Here are some rough designs I quickly mocked up,

Thinking from the mobile first, here's how I see it.
- Remove the project column all together, until there's 'space'
- Probably assuming too much here, but id imagine that the grid column for the timestamp would be `max-content`or alternatively fixed width and the left column would be `1fr`.
- Ellipsis when text to keep everything on one line. (Dirty but works for now)  

Designs below.
https://www.figma.com/file/ULqGN1mpdvUkiWJ8PCuo0z/Index-v0-%5BSpecs%5D?node-id=200%3A4719

ryan953 replied:
awesome, that's much more concrete and specific!

billyvg commented:
@Danondso can we make sure the timestamp header + sort arrow doesn't wrap? Actually -- let's spin it off into a new issue so that this PR doesn't get too bloated

untitaker commented:
unsubscribe

chadwhitacre commented:
subscribe 🍿

Danondso committed:
fix: added min-width to the headers so the timestamp column doesn't wrap
style: made the project column disappear if we're on a small screen
style: adjusted the column widths so the session column doesn't hog all the space

Danondso commented:
> @Danondso can we make sure the timestamp header + sort arrow doesn't wrap? Actually -- let's spin it off into a new issue so that this PR doesn't get too bloated

Oh shoot I did it in here, it was a one-liner change so it's not terribly obtrusive.

Jesse-Box commented:
## Feeback
### General
- Change `padding` of table body cells to `'12px 16px'`
- Center Align the items of each grid cell.
- Remove the `grid-template-column` value for viewports rider than `1440px`
### Session Column
- Link to a Replay's detail page should only wrap the username/email/IP address.
- Increase `font-size` of username/email/IP address to a computed value of `16px`
- Change `font-weight` of username/email/IP address to a computed value of `400`
- Increase `line-height` of username/email/IP address to a computed value of `1.2`
- Increase `line-height` of pathname to a computed value of `1.2`
### Time Stamp
- Decrease `gap` between Icon and timestamp to a computed value of `8px`

### Loom (For more info)
https://www.loom.com/share/14a4798f0d884d4885a6b55458348266

Danondso committed:
refactor: made the link to the replay event be the username/email/ip instead of the entire username row.
style: vertically aligned the project and timestamp columns.
refactor: removed the media query for wide screens

Danondso commented:
> ## Feeback
> ### General
> * Change `padding` of table body cells to `'12px 16px'`
> * Center Align the items of each grid cell.
> * Remove the `grid-template-column` value for viewports rider than `1440px`
> 
> ### Session Column
> * Link to a Replay's detail page should only wrap the username/email/IP address.
> * Increase `font-size` of username/email/IP address to a computed value of `16px`
> * Change `font-weight` of username/email/IP address to a computed value of `400`
> * Increase `line-height` of username/email/IP address to a computed value of `1.2`
> * Increase `line-height` of pathname to a computed value of `1.2`
> 
> ### Time Stamp
> * Decrease `gap` between Icon and timestamp to a computed value of `8px`
> 
> ### Loom (For more info)
> https://www.loom.com/share/14a4798f0d884d4885a6b55458348266

I was a little fuzzy on the padding adjustment but I added one in there. Besides that I believe I've tackled everything you've asked for in here.

ryan953 commented on static/app/views/replays/replays.tsx:
```diff
@@ -205,13 +217,23 @@ function Replays(props: Props) {
+const StyledPanelTable = styled(PanelTable)`
+  grid-template-columns: minmax(0, 1fr) max-content max-content;
+
+  @media (max-width: ${p => p.theme.breakpoints[0]}) {
```
Coupling this to the `isScreenLarge` variable, or `useMedia()` seems like it might be helpful to the next person making changes, but I can't sort out a suggestion for how.

Danondso replied:
Oh I saw that used somewhere else where a bool prop was passed into the styled component. I bet I can just pass the same media query string in, lemme see here.

ryan953 replied:
It could work like:
`<StyledPanelTable isScreenLarge>`

```
const StyledPanelTable = styled(PanelTable)<{isScreenLarge: boolean}>`
  grid-template-columns: ${
    p => p.isScreenLarge
      ? 'minmax(0, 1fr) max-content max-content'
      : 'minmax(0, 1fr) max-content'}
`;
```

but is that better? we're re-rendering anyway to drop the extra column so maybe it's not forced?

Danondso replied:
Yeah I see what you mean, I thought you could pass the string straight in and styled component would interpret it as part of the css. But that doesn't work.

```javascript
const StyledPanelTable = styled(PanelTable)<{mediaQuery: string}>`
  grid-template-columns: minmax(0, 1fr) max-content max-content;

  @media ${p => p.mediaQuery} {
    grid-template-columns: minmax(0, 1fr) max-content;
  }
`;
```

I'm good with leaving it as-is if you are, it's pretty easy to follow what's up right now I think.

ryan953 reviewed (approved):
I'm happy with things, the 1200px and smaller sizes look like they work really well now.

Danondso committed:
Merge branch 'master' into fix/33780/mobile-views-table-width-adjustment
